Code/DataStructs/RealValueVect.h

```diff
@@ -0,0 +1,134 @@
+//
+//  Copyright (c) 2014-2024, Novartis Institutes for BioMedical Research and
+//  other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <RDGeneral/export.h>
+#ifndef RD_REAL_VALUE_VECT_20140407
+#define RD_REAL_VALUE_VECT_20140407
+
+#include <boost/smart_ptr.hpp>
+#include <string>
+#include <cstring>
+
+namespace RDKit {
+
+//! a class for efficiently storing vectors of float values
+//! Has additional features compared to std::vector<double>:
+//! construct from and write to pickle
+class RDKIT_DATASTRUCTS_EXPORT RealValueVect {
+ public:
+  typedef boost::shared_array<double> DATA_SPTR;
```

Comment:
I can't see where the data is actually shared.  For instance,  In the copy constructor, it is copied.  It might be a lot easier to keep the storage class as std::vector<double> and make a thin wrapper around the extra operations.

I would also consider just using a std::vector<double> and having free functions add the extra operations, but that's a big change from the current code.

Reply:
agreed, it makes sense to make this change. I'll do it